clientlibrary/checkpoint/dynamodb-checkpointer.go

```diff
@@ -238,6 +261,18 @@ func (checkpointer *DynamoCheckpoint) FetchCheckpoint(shard *par.ShardStatus) er
 	if assignedTo, ok := checkpoint[LEASE_OWNER_KEY]; ok {
 		shard.AssignedTo = aws.StringValue(assignedTo.S)
 	}
+
+	// Use up-to-date leaseTimeout to avoid ConditionalCheckFailedException when claiming
+	if checkpointer.kclConfig.EnableLeaseStealing {
```

Comment:
No need to be specific for lease stealing.

Reply:
This is fixed.